apps/meteor/ee/app/livechat-enterprise/server/hooks/afterTakeInquiry.ts

```diff
@@ -1,25 +1,55 @@
+import type { InquiryWithAgentInfo, IOmnichannelRoom } from '@rocket.chat/core-typings';
+import { LivechatVisitors } from '@rocket.chat/models';
+
+import { RoutingManager } from '../../../../../app/livechat/server/lib/RoutingManager';
+import { afterTakeInquiry } from '../../../../../app/livechat/server/lib/hooks';
 import { settings } from '../../../../../app/settings/server';
-import { callbacks } from '../../../../../lib/callbacks';
+import { AutoTransferChatScheduler } from '../lib/AutoTransferChatScheduler';
 import { debouncedDispatchWaitingQueueStatus } from '../lib/Helper';
+import { OmnichannelQueueInactivityMonitor } from '../lib/QueueInactivityMonitor';
 import { cbLogger } from '../lib/logger';
 
-callbacks.add(
-	'livechat.afterTakeInquiry',
-	async ({ inquiry }) => {
-		if (!settings.get('Livechat_waiting_queue')) {
-			return inquiry;
-		}
+afterTakeInquiry.patch(
+	async (
+		originalFn,
+		{ inquiry, room }: { inquiry: InquiryWithAgentInfo; room: IOmnichannelRoom },
+		agent: { agentId: string; username: string },
+	) => {
+		await originalFn({ inquiry, room }, agent);
 
 		if (!inquiry) {
-			return null;
+			return;
+		}
+
+		if (settings.get('Livechat_waiting_queue')) {
+			const { department } = inquiry;
+			void debouncedDispatchWaitingQueueStatus(department);
+			cbLogger.debug({
+				msg: 'Queue status updated',
+				queue: department || 'public',
+			});
 		}
 
-		const { department } = inquiry;
-		debouncedDispatchWaitingQueueStatus(department);
+		if (settings.get<boolean>('Livechat_last_chatted_agent_routing') && agent && RoutingManager.getConfig()?.autoAssignAgent) {
+			const { v: { token } = {} } = inquiry;
+			if (token) {
+				await LivechatVisitors.updateLastAgentByToken(token, { ...agent, ts: new Date() });
+				cbLogger.info({
+					msg: 'Updated last agent of visitor',
+					token,
+					newAgent: { _id: agent.agentId, username: agent.username },
+				});
+			}
+		}
+
+		const autoTransferTimeout = settings.get<number>('Livechat_auto_transfer_chat_timeout');
+		if (autoTransferTimeout && !(room?.autoTransferredAt || room?.autoTransferOngoing)) {
```

Comment:
let's get rid of `!(conditions)` 😬 
```suggestion
		if (autoTransferTimeout && !room?.autoTransferredAt && !room?.autoTransferOngoing)) {
```

Reply:
I actually like these 🤔 
Do u think it's more readable that way?

Reply:
yes.. I have a really hard time wrapping my head around those inverted multiple conditions lol